### ISSUE TITLE
No Syntax Highlighting on v2024.14.0

### ISSUE THREAD

**Issue by axelmukwena** [microsoft/pylance-release#6358]
Type: <b>Bug</b>

# Behaviour

I'm running the latest Python extension v2024.14.0 which was released a few hours ago and I have lost syntax highlighting

## Steps to reproduce:

1. Only have Python and Pylance (auto-installed) installed
2. Go to a file.py and hover over a Python variable and no highlighting. Imports are not highlighted, no command+click just to reference, etc

<img width="1512" alt="Screenshot 2024-09-06 at 1 09 07 PM" src="https://github.com/user-attachments/assets/d79b3bba-0393-4b4b-814f-07860a74f231">

3. If you downgrade, let's say to `v2024.12.3`, everything will work as expected

<img width="1512" alt="Screenshot 2024-09-06 at 1 08 47 PM" src="https://github.com/user-attachments/assets/0bd5375d-04ea-4029-9490-f0e133d26abc">

# Diagnostic data

Nothing in the output panel

Extension version: 2024.14.0
VS Code version: Code 1.89.1 (Universal) (dc96b837cf6bb4af9cd736aa3af08cf8279f7685, 2024-05-07T05:14:24.611Z)
OS version: Darwin arm64 23.4.0
Modes:

-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.11
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): Venv
-   Value of the `python.languageServer` setting: Default

<details>
<summary>User Settings</summary>
<p>

```

languageServer: "Pylance"

```
</p>
</details>

<details>
<summary>Installed Extensions</summary>

|Extension Name|Extension Id|Version|
|---|---|---|
|JavaScript Debugger|ms-vscode.js-debug|1.89.0|
|JavaScript Debugger Companion Extension|ms-vscode.js-debug-companion|1.1.2|
|Pylance|ms-python.vscode-pylance|2024.8.2|
|Python|ms-python.python|2024.14.0|
|Table Visualizer for JavaScript Profiles|ms-vscode.vscode-js-profile-table|1.0.9|
</details>
<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M3 Pro (11 x 24)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|18.00GB (0.25GB free)|
|Process Argv|--crash-reporter-id ad33eb45-3e2b-41ab-882a-2c8d90996c0a|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
bdiig495:31013172
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
impr_priority:31102340
refactort:31108082
pythonrstrctxt:31112756
wkspc-onlycs-t:31132770
wkspc-ranged-t:31125599
fje88620:31121564

```

</details>

**Comment by rchiodo** [comment]
There should be an error in the `Python Language Server` output that says this:

```
2024-09-06 09:37:45.381 [info] The language client requires VS Code version ^1.91.0 but received version 1.89.1
```

You would need to upgrade VS code to the latest version to use the newer Python/Pylance extension.

**Comment by rchiodo** [comment]
Internally we need to mark our engine version for both extensions to 1.91.

**Comment by IamRezaMousavi** [comment]
@rchiodo I have similar bug:

As you can see, all libraries and variables are used:

![x](https://github.com/user-attachments/assets/541b5de5-20e0-4ce1-8dc4-ed376a8e05be)
![y](https://github.com/user-attachments/assets/44a55130-84cf-4a4b-ab91-fb09971c6e99)

**Comment by IamRezaMousavi** [comment]
> > [@rchiodo](https://github.com/rchiodo) I have similar bug:
> > As you can see, all libraries and variables are used:
> > ![x](https://private-user-images.githubusercontent.com/83984878/365892690-541b5de5-20e0-4ce1-8dc4-ed376a8e05be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwMzI0MDQsIm5iZiI6MTcyNjAzMjEwNCwicGF0aCI6Ii84Mzk4NDg3OC8zNjU4OTI2OTAtNTQxYjVkZTUtMjBlMC00Y2UxLThkYzQtZWQzNzZhOGUwNWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDA1MjE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3MTkwMzI0M2E5YzVlOWU0MDlkZThhMTA5Mzk1Y2Y0MzdjZTIwOWJjYzBiNGI3ZWQ4MGJiMWE4OGYyYmUwZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pOi8Qd2UwDsEmG5utF9e3s-UvDQbAKsdXwdrEBKyzuk) ![y](https://private-user-images.githubusercontent.com/83984878/365892715-44a55130-84cf-4a4b-ab91-fb09971c6e99.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwMzI0MDQsIm5iZiI6MTcyNjAzMjEwNCwicGF0aCI6Ii84Mzk4NDg3OC8zNjU4OTI3MTUtNDRhNTUxMzAtODRjZi00YTRiLWFiOTEtZmIwOTk3MWM2ZTk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDA1MjE0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQwZjc3MTEzNGNlOWU2NDgzYWRhNDg5ZjliMjM2Mzg2Y2FlMDk3ZWUyY2YxNjQyMjA0MGY1OTI3ZWFkNWNjYTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NGk0fw2ZiXTZ9GLEwr9EpXu0EOecZw87MtQg8oNXUK0)
> 
> @IamRezaMousavi , I don't think your issue is the same. The root cause of the original issue is that Pylance requires the latest VS code now. I'm not sure what your problem is. It looks correct to me. Colorization is different for assignment vs usage.

- imported socket in line 3 and used in line 21
- assigned udp_port in line 19 and used in line 22
- assigned addr in line 28 and used in line 29
- assigned data_dict in line 34 and used in line 38
- And many other...

These items have wrong colors

**Comment by rchiodo** [comment]
I'll open a separate issue. It's still not the same as this one.

**Comment by StellaHuang95** [comment]
This issue has been fixed in prerelease version 2024.9.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20249100-12-September-2024-PreRelease)

**Comment by axelmukwena** [comment]
Thank you @StellaHuang95 🥬
